inst/eDISH_app/server.R

```diff
@@ -28,37 +28,84 @@ function(input, output, session){
   #
   # reutrns updated settings and validation status
     settings_new <-   callModule(renderSettings, "settingsUI",
-                                # data = isolate(reactive(dataUpload_out$data_selected())),  # this doesnt make sense
                                  data = reactive(dataUpload_out$data_selected()),
                                  settings = reactive(dataUpload_out$settings()),
                                  status = reactive(dataUpload_out$status()))
 
 
   # update settings navbar
     output$settings_tab_title = renderUI({
-      if (settings_new$status()$valid==TRUE){
+      num_valid <- settings_new$status() %>% flatten() %>%  keep(., names(.)=="valid") %>% unlist() %>% sum()
+    #  if (settings_new$status()$valid==TRUE){
+      if(num_valid >0) {
         HTML(paste("Settings", icon("check", class="ok")))
       } else {
         HTML(paste("Settings", icon("times", class="notok")))
       }
     })
 
     # update charts navbar
-    output$chart_tab_title = renderUI({
-      if (settings_new$status()$valid==TRUE){
-        HTML(paste("Chart", icon("check", class="ok")))
-      } else {
-        HTML(paste("Chart", icon("times", class="notok")))
-      }
-    })
+    # output$chart_tab_title = renderUI({
+    #   if (settings_new$status()$valid==TRUE){
+    #     HTML(paste("Chart", icon("check", class="ok")))
+    #   } else {
+    #     HTML(paste("Chart", icon("times", class="notok")))
+    #   }
+    # })
+
+   # ## this currently wipes away everything anytime there's a change in chart selections OR
+    #  change in validation status
```

Comment:
As discussed, this probably needs a refactor ... Not sure if it's needed this release though. Let's decide in our next check-in.